diff -ru gnutls-3.8.11.orig/lib/audit.c gnutls-3.8.11/lib/audit.c
--- gnutls-3.8.11.orig/lib/audit.c	2025-11-19 06:53:30.000000000 +0100
+++ gnutls-3.8.11/lib/audit.c	2025-11-22 14:25:28.617946888 +0100
@@ -25,7 +25,7 @@
 #define CRAU_CONTEXT_STACK_DEPTH 8
 #include "audit.h"
 
-static _Thread_local struct crau_context_stack_st stack;
+static __thread struct crau_context_stack_st stack;
 
 /**
  * gnutls_audit_push_context:
diff -ru gnutls-3.8.11.orig/lib/crau/crau.h gnutls-3.8.11/lib/crau/crau.h
--- gnutls-3.8.11.orig/lib/crau/crau.h	2025-10-24 08:10:22.000000000 +0200
+++ gnutls-3.8.11/lib/crau/crau.h	2025-11-22 14:24:15.093926641 +0100
@@ -251,12 +251,7 @@
 # else
 
 #  ifndef CRAU_MAYBE_UNUSED
-#   if defined(__has_c_attribute) && \
-    __has_c_attribute (__maybe_unused__)
-#    define CRAU_MAYBE_UNUSED [[__maybe_unused__]]
-#   elif defined(__GNUC__)
 #    define CRAU_MAYBE_UNUSED __attribute__((__unused__))
-#   endif
 #  endif /* CRAU_MAYBE_UNUSED */
 
 void crau_push_context(struct crau_context_stack_st *stack CRAU_MAYBE_UNUSED,
diff -ru gnutls-3.8.11.orig/lib/system/sockets.c gnutls-3.8.11/lib/system/sockets.c
--- gnutls-3.8.11.orig/lib/system/sockets.c	2025-09-25 08:22:49.000000000 +0200
+++ gnutls-3.8.11/lib/system/sockets.c	2025-11-22 14:17:14.306899955 +0100
@@ -161,7 +161,7 @@
 ssize_t system_writev(gnutls_transport_ptr_t ptr, const giovec_t *iovec,
 		      int iovec_cnt)
 {
-	return _system_writev(ptr, iovec, iovec_cnt, 0);
+	return writev(GNUTLS_POINTER_TO_INT(ptr), (struct iovec *) iovec, iovec_cnt);
 }
 
 #endif
