diff -ru gnutls-3.8.12.orig/lib/audit.c gnutls-3.8.12/lib/audit.c
--- gnutls-3.8.12.orig/lib/audit.c	2026-02-09 15:41:57.000000000 +0100
+++ gnutls-3.8.12/lib/audit.c	2026-04-17 19:18:14.698414836 +0200
@@ -25,7 +25,7 @@
 #define CRAU_CONTEXT_STACK_DEPTH 8
 #include "audit.h"
 
-static _Thread_local struct crau_context_stack_st stack;
+static __thread struct crau_context_stack_st stack;
 
 /**
  * gnutls_audit_push_context:
diff -ru gnutls-3.8.12.orig/lib/random.c gnutls-3.8.12/lib/random.c
--- gnutls-3.8.12.orig/lib/random.c	2026-02-09 15:41:57.000000000 +0100
+++ gnutls-3.8.12/lib/random.c	2026-04-17 19:25:55.012924015 +0200
@@ -55,7 +55,7 @@
 
 /* Flag to indicate initialization
  */
-static _Thread_local unsigned rnd_initialized = 0;
+static __thread unsigned rnd_initialized = 0;
 
 static void free_ctx(const void *ctx)
 {
diff -ru gnutls-3.8.12.orig/lib/system/sockets.c gnutls-3.8.12/lib/system/sockets.c
--- gnutls-3.8.12.orig/lib/system/sockets.c	2026-02-09 15:41:57.000000000 +0100
+++ gnutls-3.8.12/lib/system/sockets.c	2026-04-17 19:18:14.698851859 +0200
@@ -161,7 +161,7 @@
 ssize_t system_writev(gnutls_transport_ptr_t ptr, const giovec_t *iovec,
 		      int iovec_cnt)
 {
-	return _system_writev(ptr, iovec, iovec_cnt, 0);
+	return writev(GNUTLS_POINTER_TO_INT(ptr), (struct iovec *) iovec, iovec_cnt);
 }
 
 #endif
