diff -ru gnutls-3.8.11.orig/lib/audit.c gnutls-3.8.11/lib/audit.c
--- gnutls-3.8.11.orig/lib/audit.c	2025-11-19 06:53:30.000000000 +0100
+++ gnutls-3.8.11/lib/audit.c	2025-11-22 14:25:28.617946888 +0100
@@ -25,7 +25,7 @@
 #define CRAU_CONTEXT_STACK_DEPTH 8
 #include "audit.h"
 
-static _Thread_local struct crau_context_stack_st stack;
+static __thread struct crau_context_stack_st stack;
 
 /**
  * gnutls_audit_push_context:
diff -ru gnutls-3.8.11.orig/lib/system/sockets.c gnutls-3.8.11/lib/system/sockets.c
--- gnutls-3.8.11.orig/lib/system/sockets.c	2025-09-25 08:22:49.000000000 +0200
+++ gnutls-3.8.11/lib/system/sockets.c	2025-11-22 14:17:14.306899955 +0100
@@ -161,7 +161,7 @@
 ssize_t system_writev(gnutls_transport_ptr_t ptr, const giovec_t *iovec,
 		      int iovec_cnt)
 {
-	return _system_writev(ptr, iovec, iovec_cnt, 0);
+	return writev(GNUTLS_POINTER_TO_INT(ptr), (struct iovec *) iovec, iovec_cnt);
 }
 
 #endif
