diff -ru ncdu-1.22.orig/src/dir_scan.c ncdu-1.22/src/dir_scan.c
--- ncdu-1.22.orig/src/dir_scan.c	2025-02-05 20:37:02.000000000 +0100
+++ ncdu-1.22/src/dir_scan.c	2026-04-07 20:07:41.567577008 +0200
@@ -38,10 +38,12 @@
 #include <sys/attr.h>
 #endif
 
-#if HAVE_LINUX_MAGIC_H && HAVE_SYS_STATFS_H && HAVE_STATFS
+#if HAVE_SYS_STATFS_H && HAVE_STATFS
 #include <sys/statfs.h>
+#if HAVE_LINUX_MAGIC_H
 #include <linux/magic.h>
 #endif
+#endif
 
 
 /* set S_BLKSIZE if not defined already in sys/stat.h */
@@ -61,7 +63,7 @@
 static unsigned int buf_nlink;
 
 
-#if HAVE_LINUX_MAGIC_H && HAVE_SYS_STATFS_H && HAVE_STATFS
+#if HAVE_LINUX_MAGIC_H && HAVE_SYS_STATFS_H && HAVE_STATFS || __midipix__
 
 static int is_kernfs(unsigned long type) {
   if(
diff -ru ncdu-1.22.orig/src/main.c ncdu-1.22/src/main.c
--- ncdu-1.22.orig/src/main.c	2025-03-03 13:39:30.000000000 +0100
+++ ncdu-1.22/src/main.c	2026-04-07 20:06:45.772077370 +0200
@@ -315,7 +315,7 @@
   "  -X, --exclude-from FILE    Exclude files that match any pattern in FILE\n"
   "  --exclude-caches           Exclude directories containing CACHEDIR.TAG\n"
   "  -L, --follow-symlinks      Follow symbolic links (excluding directories)\n"
-#if HAVE_LINUX_MAGIC_H && HAVE_SYS_STATFS_H && HAVE_STATFS
+#if HAVE_LINUX_MAGIC_H && HAVE_SYS_STATFS_H && HAVE_STATFS || __midipix__
   "  --exclude-kernfs           Exclude Linux pseudo filesystems (procfs,sysfs,cgroup,...)\n"
 #endif
 #if HAVE_SYS_ATTR_H && HAVE_GETATTRLIST && HAVE_DECL_ATTR_CMNEXT_NOFIRMLINKPATH
@@ -441,9 +441,11 @@
     else if(!arg_option(0)) die("Unknown option '%s'.\n", argparser_state.last);
   }
 
+#ifndef __midipix__
 #if !(HAVE_LINUX_MAGIC_H && HAVE_SYS_STATFS_H && HAVE_STATFS)
   if(exclude_kernfs) die("The --exclude-kernfs flag is currently only supported on Linux.\n");
 #endif
+#endif
 
   if(export) {
     if(dir_export_init(export)) die("Can't open %s: %s\n", export, strerror(errno));
